src/modules/gimbal/output.cpp

```diff
@@ -81,6 +83,27 @@ float OutputBase::_calculate_pitch(double lon, double lat, float altitude,
 	return atan2f(z, target_distance);
 }
 
+bool OutputBase::check_and_handle_setpoint_timeout(ControlData &control_data, const hrt_abstime &now)
+{
+	bool ret = false;
+	const bool timeout = (control_data.timestamp_last_update + 2_s < now);
+	const bool type_angle = (control_data.type == ControlData::Type::Angle);
+
+	if (timeout && type_angle) {
+		// Avoid gimbal keeps on spinning if the last setpoint was angular_velocity but it times out
+		for (int i = 0; i < 3; ++i) {
+			const float vel = control_data.type_data.angle.angular_velocity[i];
+
+			if (PX4_ISFINITE(vel) && (fabsf(vel) > FLT_EPSILON)) {
+				control_data.type_data.angle.angular_velocity[i] = 0.f;
+				ret = true;
```

Comment:
This is an alias, right? So you don't need to copy it:

```suggestion
			float &vel = control_data.type_data.angle.angular_velocity[i];

			if (PX4_ISFINITE(vel) && (fabsf(vel) > FLT_EPSILON)) {
				vel = 0.f;
				ret = true;
```

Reply:
Yes, that's correct. I'd assume the copy is optimized away. Your suggestion might be more readable. I don't have a strong preference and will change it 👍

Reply:
I think you're right. It seems to get optimized away indeed no matter what:
https://godbolt.org/z/Wc5515W97

Reply:
Thanks for verifying, that's good to know 👍
I still changed it like you suggest, will push in a minute.

Reply:
Put in place: cb7a83f